models/Asset/Image/Thumbnail/Processor.php

```diff
@@ -472,6 +475,27 @@ public static function process(Asset $asset, Config $config, $fileSystemPath = n
         ];
     }
 
+    /**
+     * @throws \Exception
+     */
+    private static function addOrUpdateThumbnailCache(
+        Asset $asset,
+        Config $config,
+        string $filename
+    ): void
+    {
+        $tmpFsPath = $asset->getLocalFile();
```

Comment:
I think we need to move this to a public service method, because we need to consider the changes made in #15323 as well. 
Because actually `getimagesize()` would fail here as well for AVIF images.

Reply:
changed visibility of `readDimensionsFromFile` to public and marked as internal and this is then called in the processor now.